web/entities/cluster.go

```diff
@@ -14,24 +15,120 @@ type Cluster struct {
 	SIDs            pq.StringArray `gorm:"column:sids; type:text[]"`
 	ResourcesNumber int
 	HostsNumber     int
-	Tags            []models.Tag `gorm:"polymorphic:Resource;polymorphicValue:clusters"`
+	Tags            []*models.Tag `gorm:"polymorphic:Resource;polymorphicValue:clusters"`
 	UpdatedAt       time.Time
+	Hosts           []*Host        `gorm:"foreignkey:cluster_id"`
+	Details         datatypes.JSON `json:"payload" binding:"required"`
 }
 
-func (h *Cluster) ToModel() *models.Cluster {
+type HANAClusterDetails struct {
+	SystemReplicationMode          string             `json:"system_replication_mode"`
+	SystemReplicationOperationMode string             `json:"system_replication_operation_mode"`
+	SecondarySyncState             string             `json:"secondary_sync_state"`
+	SRHealthState                  string             `json:"sr_health_state"`
+	CIBLastWritten                 time.Time          `json:"cib_last_written"`
+	StonithType                    string             `json:"stonith_type"`
+	StoppedResources               []*ClusterResource `json:"stopped_resources"`
+	Nodes                          []*ClusterNode     `json:"nodes"`
+	SBDDevices                     []*SBDDevice       `json:"sbd_devices"`
```

Comment:
I would say that the `SBDDevices` should be referenced in the `Cluster` entity, and not in the details. All sort of clusters have sbd devices. Netweaver will have sbd devices

Reply:
Alright, but it is still part of the details field so we will duplicate the nesting also in the other clusters, i wouldn't put this in the generics fields